### ISSUE TITLE
Handshake only consideres the highest matching version

### ISSUE THREAD

**Issue by karknu** [input-output-hk/ouroboros-network#2695]
If the client and server disagree on a protocol parameter for the highest matching version the server rejects the handshake instead of considering older versions.

**Comment by coot** [comment]
I agree; this was ok, until we didn't had any other option than the network magic.
